chemprop/nn/loss.py

```diff
@@ -149,8 +151,23 @@ def forward(self, preds: Tensor, targets: Tensor, *args) -> Tensor:
 
         return L_nll + self.v_kl * (L_reg - self.eps)
 
-    def get_params(self) -> list[tuple[str, float]]:
-        return [("v_kl", self.v_kl), ("eps", self.eps)]
+    def get_params(self, deep: bool = True) -> dict[str, float]:  # pylint: disable=unused-argument
+        """Get the parameters of the class.
+
+        Overrides the `ReprMixin.get_params` method to return the `v_kl` and `eps` parameter.
+
+        Parameters
+        ----------
+        deep: bool, default=True
+            No effect on this method.
+
+        Returns
+        -------
+        dict[str, float]
+            The parameters of the class.
+        """
+        param_dict = {"v_kl": self.v_kl, "eps": self.eps}
+        return param_dict
```

Comment:
The original docstring is still correct--there's no use in overriding it.
```suggestion
        return {"v_kl": self.v_kl, "eps": self.eps}
```

Reply:
The parent docstring reads:
```
        deep : bool, default=True
            If True, will return the parameters for this estimator and
            contained subobjects that are estimators.
```
This is a no-op for an object that contains no child objects (i.e., this one.) Because it's a no-op for this object **does not** mean it's no longer correct.

Reply:
>  A broader question: Why is `get_params()` overwritten for `EvidentialLoss`, `DirichletMixin`, and `_ThresholdMixin` anyway?
> If they produce essentially the same output, does overwriting it have a different purpose?

I believe this is a vestige of the original implementation, where I could not rely on the default implementation to work in some cases, so I had to override it here to ensure that the correct representation was returned. It now needs to be overridden in cases where the input parameter name _does not_ match the internal attribute name, i.e.:
```python
class Foo:
    def __init__(self, a: int = 5):
        self.b = a
```
In both the current and proposed implementations, we have the following:
```python
>>> Foo()
Foo()
>>> Foo(10)
Foo()
```
even though it should read:
```python
>>> Foo(10)
Foo(a=10)
```
Fixing this however would require some changes and that brings this PR out-of-sync with the `sklearn` implementation (because ours would be a general solution to the representation problem). I don't personally care about this issue TBH as long as there's a note that the class only works when parameter names match their corresponding attribute names.